docs/kubeadm.md

```diff
@@ -0,0 +1,249 @@
+For example:
+
+    # kubeadm join --token <token> <master-ip>
+    <util/tokens> validating provided token
+    <node/discovery> created cluster info discovery client, requesting info from "http://138.68.156.129:9898/cluster-info/v1/?token-id=0f8588"
+    <node/discovery> cluster info object received, verifying signature using given token
+    <node/discovery> cluster info signature and contents are valid, will use API endpoints [https://138.68.156.129:443]
+    <node/csr> created API client to obtain unique certificate for this node, generating keys and certificate signing request
+    <node/csr> received signed certificate from the API server, generating kubelet configuration
+    <util/kubeconfig> created "/etc/kubernetes/kubelet.conf"
+
+    Node join complete:
+    * Certificate signing request sent to master and response
+      received.
+    * Kubelet informed of new secure connection details.
+
+    Run 'kubectl get nodes' on the master to see this machine join.
+
+A few seconds later, you should notice that running `kubectl get nodes` on the master shows a cluster with as many machines as you created.
+
+**YOUR CLUSTER IS NOT READY YET!**
+
+Before you can deploy applications to it, you need to install a pod network.
+
+### (4/4) Installing a pod network
+
+You must install a pod network add-on so that your pods can communicate with eachother when they are on different hosts.
+
+Several projects provide Kubernetes pod networks.
+A simple one with no infrastructure or database dependencies is Weave Net, which you can install by running, on the master:
+
+    # kubectl apply -f https://git.io/weave-kube
+    daemonset "weave-net" created
+
+**You should run this on the master before you try to deploy any applications to your cluster.**
+
+Once the command has completed, a few seconds later you should see the `weave-net` pods and the `kube-dns` pod go into `Running` in the output of `kubectl get pods --all-namespaces`. **This signifies that your cluster is ready.**
+
+You can learn more about Weave for Kubernetes on the project's [GitHub page](https://github.com/weaveworks/weave-kube).
+
+You can see a complete list of available network add-ons on the [add-ons page](/docs/admin/addons/).
+
+### (Optional) Installing a sample application
+
+As an example, you will now install a sample microservices application, a socks shop, to put your cluster through its paces.
+To learn more about the sample microservices app, see the [GitHub README](https://github.com/microservices-demo/microservices-demo).
+
+Here you will install the NodePort version of the Socks Shop, which doesn't depend on Load Balancer integration, since our cluster doesn't have that:
+
+    # kubectl apply -f https://raw.githubusercontent.com/lukemarsden/microservices-demo/master/deploy/kubernetes/definitions/wholeWeaveDemo-NodePort.yaml
```

Comment:
The socks shop is not weave-specific, it's a generic microservices demo that will work on any kubernetes cluster (and lots of other things too) and lives in a [vendor-independent org on github](https://github.com/microservices-demo). I need to make a PR to add a NodePort version that will work with this guide and then I can replace the `lukemarsden` URL with one that links there instead.